Backpack-SwiftUI/FieldSet/Classes/BPKFieldSet.swift

```diff
@@ -0,0 +1,195 @@
+    private let description: String?
+    private var accessibilityPrefix: String?
+    
+    public init(
+        label: String? = nil,
+        description: String? = nil,
+        content: () -> Content
+    ) {
+        self.label = label
+        self.description = description
+        self.content = content()
+    }
+    
+    public var body: some View {
+        VStack(alignment: .leading, spacing: .sm) {
+            labelView
+            content
+                .inputState(state)
+                .padding(.bottom, .sm)
+                .accessibilityIdentifier(accessibilityIdentifier(for: "wrapped_view"))
+            descriptionView
+            if case let .error(message) = state {
+                errorMessage(message)
+                    .padding(.top, .sm)
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private var labelView: some View {
+        if let label {
+            BPKText(label, style: .label2)
+                .lineLimit(nil)
+                .foregroundColor(state.labelColor)
+                .accessibilityIdentifier(accessibilityIdentifier(for: "label"))
+        }
+    }
+    
+    @ViewBuilder
+    private var descriptionView: some View {
+        if let description {
+            BPKText(description, style: .caption)
+                .lineLimit(nil)
+                .foregroundColor(state.descriptionColor)
+                .accessibilityIdentifier(accessibilityIdentifier(for: "descritpion"))
+        }
+    }
+    
+    private func errorMessage(_ message: String) -> some View {
+        HStack(spacing: .md) {
+            BPKIconView(.exclamationCircle)
+                .foregroundColor(.textErrorColor)
+                .accessibilityHidden(true)
+            BPKText(message, style: .caption)
+                .lineLimit(nil)
+                .foregroundColor(.textErrorColor)
+                .accessibilityIdentifier(accessibilityIdentifier(for: "error_message"))
+        }
+    }
+    
+    public func inputState(_ state: BPKFieldSet<Content.WrappedView>.State) -> BPKFieldSet {
+        var result = self
+        result.state = state
+        return result
+    }
+}
+
+// MARK: - Accessibility
+
+extension BPKFieldSet {
+    public func accessibilityPrefix(_ prefix: String?) -> BPKFieldSet {
+        var result = self
+        result.accessibilityPrefix = prefix
+        return result
+    }
+    
+    private func accessibilityIdentifier(for label: String) -> String {
+        if let prefix = accessibilityPrefix {
+            return "\(prefix)_\(label)"
+        }
+        return ""
+    }
+}
+
+// MARK: - Previews
+
+#Preview("Default State") {
+    ScrollView {
+        constructViews(inErrorState: false)
+        .padding()
+    }
+}
+
+#Preview("Error State") {
+    ScrollView {
+        constructViews(inErrorState: true)
+        .padding()
+    }
+}
+
+// swiftlint:disable closure_body_length
+@ViewBuilder
+private func constructViews(inErrorState: Bool) -> some View {
```

Comment:
this code should not be here, it should be contained within the previows. here it's accessible from the component and that should not be the case.

you could create only one #preview and then assigne names to the 2 scroll views it'd have inside!